tensorflow/compiler/xla/service/algebraic_simplifier.cc

```diff
@@ -2520,6 +2523,13 @@ Status AlgebraicSimplifierVisitor::HandleTranspose(HloInstruction* transpose) {
                                            transpose->dimensions())));
   }
 
+  if (ShapeUtil::TrueRank(transpose->shape()) <= 1) {
+    HloInstruction* reshape = computation_->AddInstruction(
+        HloInstruction::CreateReshape(transpose->shape(), operand));
+    TF_RETURN_IF_ERROR(ReplaceInstruction(transpose, reshape));
+    return HandleReshape(reshape);
```

Comment:
algebraic simplifier runs in a fixed point pass, there is not a reason to recurse here?